models/gpg_key.go

```diff
@@ -239,6 +239,16 @@ func parseGPGKey(ownerID int64, e *openpgp.Entity) (*GPGKey, error) {
 	}, nil
 }
 
+// deleteGPGKeys does the actual key deletion but does not update authorized_keys file.
```

Comment:
Can you add for information that this method will not delete any related (subkey) compared to deleteGPGKey ? Something like :
```suggestion
// deleteGPGKeys does the actual key deletion but does not update authorized_keys file nor delete related key (subkey).
``` 
or add :
`_, err := e.In("primary_key_id", keyIDs).Delete(new(GPGKey))`
each solution is fine as the list passed should already contained any related keys in the original use case.

Reply:
But deleteGPGKeys does also delete all subkeys of a master key which are assigned to a user. More specifically: deleteGPGKeys deletes all keys of a specified user.

Reply:
This is just in case for later use in other part in code. Currently when you call this method [here](https://github.com/go-gitea/gitea/pull/5357/files#diff-46259196476f860fea33754fcb22e9eeR1065 ) you pass all keys (add subkeys) owned by user. But if this method is used somewhere else there could be orphaned subkeys if not all the key are passed as argument. This is just to add a warning for later use (or add the delete of any sub key).